R/exclude.R

```diff
@@ -25,36 +29,54 @@ exclude <- function(lints, exclusions = settings$exclusions, ...) {
 
 
   exclusions <- normalize_exclusions(c(source_exclusions, exclusions))
-  to_exclude <- vapply(seq_len(nrow(df)),
+  to_exclude <- vapply(
+    seq_len(nrow(df)),
     function(i) {
       file <- df$filename[i]
       file %in% names(exclusions) &&
-        length(exclusions[[file]]) == 1 &&
-        exclusions[[file]] == Inf ||
-        df$line_number[i] %in% exclusions[[file]]
-     },
-    logical(1))
+        is_excluded(df$line_number[i], df$linter[i], exclusions[[file]])
+    },
+    logical(1L)
+  )
 
   if (any(to_exclude)) {
     lints <- lints[!to_exclude]
   }
 
   lints
 }
+
+is_excluded <- function(line_number, linter, file_exclusion) {
+  excluded_lines <- unlist(file_exclusion[names2(file_exclusion) %in% c("", linter)])
+  Inf %in% excluded_lines || line_number %in% excluded_lines
```

Comment:
I would use `any(is.infinite(excluded_lines))`

Reply:
The docs only permit `Inf` as a sentinel. Do we want to (silently) allow `-Inf` as well?

Reply:
I see... `any(excluded_lines == Inf)` doesn't look much better and is probably inefficient. Fine as is.